### PR TITLE
Feature: Member profile table and individual view

### PR DESCRIPTION
**Pull request by roselynh100**
**PLEASE MERGE SAMEEKSHA'S PR FIRST, I PULLED FROM HERS AND EDITED IT**

**new features:**
- typescript react babeyyyy
- edited backend stuff... somewhere... to make it actually work for me (`__init__.py`)
- added the whole frontend so far:
  - basic layout of the website with space for navbar and header
  - member list (in horizontal card format)
  - member profile (in a modal, when you click a member from the list)
- member data actually gets loaded in from backend - all currently available fields work except "shirt size" because i forgot about that lol

**next steps:**
- make member profile "view mode" first, then have an edit button to allow changes (i don't think this is SUPER high priority though, since it's an internal tool)
- allow creation and editing of members from the frontend (needs backend to have this set up first)

image
image

### CODE REVIEW

**Review by hydrofluorich — CHANGES_REQUESTED** [review]
~~Per [CL-24](https://uwcbc.atlassian.net/browse/CL-24), shirt size isn't a field that we're currently including in Celesta. From what I can tell, it's currently included in the frontend but not the backend? (Correct me if I'm wrong.) Will bring this up in Discord, wait until a decision to merge.~~
UPDATE: I was wrong. Shirt size currently exists in the backend. New ticket will be created to fix it in the frontend.

File: frontend/src/components/MemberProfile.tsx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+const faculties = [
+	'Arts',
+	'Engineering',
+	'Environment',
+	'Health',
+	'Math',
+	'Science'
+]
+
+const instruments = [
```
Comment (hydrofluorich) [inline frontend/src/components/MemberProfile.tsx]: Can we change these either to instruments or to be empty for now? And add a TODO for [CL-42](https://uwcbc.atlassian.net/browse/CL-42)

File: frontend/src/components/MemberEntry.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { Box, Flex, Group, Image, Text } from '@mantine/core'
+import { Member } from 'types'
+
+interface MemberEntryProps {
+	member: Member
+	image?: string
+}
+
+// TODO: link member.mainInstrument to instrument assets, and add created time
```
Comment (hydrofluorich) [inline frontend/src/components/MemberEntry.tsx]: Add note that this is for [CL-42](https://uwcbc.atlassian.net/browse/CL-42)

File: frontend/src/types.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+export interface Member {
+	id: string
+	first_name: string
+	last_name: string
+	email: string
+	member_type: number
+	faculty?: string
+	studentNumber?: number
+	watIAM?: string | null
+	mainInstrument: string
```
Comment (hydrofluorich) [inline frontend/src/types.ts:10]: Does this have to be a mandatory field?

Comment (roselynh100) [reply frontend/src/types.ts:10]: updated to be consistent with the other non-mandatory (uw student) fields

**Review by s-naik — APPROVED** [review]
I'll have to keep an eye on the changes made to backend/__init__.py, but if it works for Roselyn now then I think it's fine to roll with it.

**Review by hydrofluorich — APPROVED** [review]

**Pull request merged**